src/components/AppComponents/LanguageToggle/index.tsx

```diff
@@ -8,24 +8,24 @@ const LanguageToggle = (props) => {
   const { i18n } = useTranslation();
   let lang = sessionStorage.getItem('rsdk_locale')?.substring(0, 2) || 'en';
   const [selectedLang, setSelectedLang] = useState(lang);
-  // i18n.changeLanguage(lang);
 
-  const changeLanguage = e => {
+  const changeLanguage = (e) => {
     e.preventDefault();
     lang = e.currentTarget.getAttribute('lang');
     setSelectedLang(lang);
     sessionStorage.setItem('rsdk_locale', `${lang}_GB`);
     i18n.changeLanguage(lang);
     if (PegaApp) {
       PCore.getEnvironmentInfo().setLocale(`${lang}_GB`);
-
-      // This cleares the generic fields localisation values and attempts to 'refetch' them, as they do not do this
-      // manually after locale is updated
       PCore.getLocaleUtils().resetLocaleStore();
       PCore.getLocaleUtils().loadLocaleResources([PCore.getLocaleUtils().GENERIC_BUNDLE_KEY, '@BASECLASS!DATAPAGE!D_LISTREFERENCEDATABYTYPE']);
     }
   };
 
+  useEffect(() => {
```

Comment:
can we have useEffect on top of file?

Reply:
Placing useEffect hook in any component depends on the desired behaviour. In the case above since I want to update the HTML lang attr based on changes to the selectLang state, it makes more sense to have the useEffect after changeLanguage function . not before it :)  Thanks @gautamkumar-hmrc